### PR TITLE
attach claim files

### PR DESCRIPTION
**Pull request by Schparky**

### CODE REVIEW

File: application/actions/actions.go [hunk 1]
```diff
@@ -38,7 +39,8 @@ func StrictBind(c buffalo.Context, dest interface{}) error {
 // reportError logs an error with details and renders the error with buffalo.Render.
 // If the HTTP status code provided is in the 300 family, buffalo.Redirect is used instead.
 func reportError(c buffalo.Context, err error) error {
-	appErr, ok := err.(*api.AppError)
+	var appErr *api.AppError
+	ok := errors.As(err, &appErr)
```
Comment (jason-jackson) [inline application/actions/actions.go:43]: Could simplify to `if !errors.As(err, &appErr) {` and remove following line

File: application/models/models.go [hunk 2]
```diff
@@ -141,6 +141,26 @@ func create(tx *pop.Connection, m interface{}) error {
 	return nil
 }
 
+func appErrorFromDB(err error, defaultKey api.ErrorKey) error {
+	appErr := api.NewAppError(err, defaultKey, api.CategoryInternal)
+
+	var pgError *pgconn.PgError
+	if ok := errors.As(err, &pgError); ok {
```
Comment (jason-jackson) [inline application/models/models.go:148]:
```suggestion
	if errors.As(err, &pgError) {
```

File: application/models/models.go [hunk 3]
```diff
@@ -141,6 +141,26 @@ func create(tx *pop.Connection, m interface{}) error {
...
+		appErr.Err = fmt.Errorf("%w Detail: %s", pgError, pgError.Detail)
+
+		switch pgError.Code {
+		case "23503":
```
Comment (jason-jackson) [inline application/models/models.go:152]: Do we want to use github.com/jackc/pgerrcode to get error codes rather than hardcoding?

**Review by jason-jackson — APPROVED** [review]

File: application/models/claimfile.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+	ID        uuid.UUID `db:"id"`
+	ClaimID   uuid.UUID `db:"claim_id" validate:"required"`
+	FileID    uuid.UUID `db:"file_id" validate:"required"`
+	CreatedAt time.Time `db:"created_at"`
+	UpdatedAt time.Time `db:"updated_at"`
+}
+
+type ClaimFiles []ClaimFile
+
+// NewClaimFile maken a new ClaimFile but does not save it to the database
```
Comment (Baggerone) [inline application/models/claimfile.go:24]: *makes

File: application/models/file.go [hunk 5]
```diff
@@ -102,19 +102,19 @@ func (f *File) Store(tx *pop.Connection) *FileUploadError {
...
+			Message:    fmt.Sprintf("error %s storing file: %+v", err, f),
 		}
 		return &e
 	}
 
 	f.URL = url.Url
 	f.URLExpiration = url.Expiration
 	f.Size = len(f.Content)
-	if err := f.Create(tx); err != nil {
+	if err = f.Create(tx); err != nil {
```
Comment (Baggerone) [inline application/models/file.go:113]: I'm curious about your reason for making this change.

**Review by Baggerone — APPROVED** [review]

File: application/models/claimfile.go (hunk 4 above)
Comment (jason-jackson) [reply application/models/claimfile.go:24]: *maketh

File: application/models/models.go (hunk 3 above)
Comment (Schparky) [reply application/models/models.go:152]: Good find, thanks. I looked (briefly) for such a library but only found one for javascript. Seems this one was right under my nose.

**Pull request merged**